presto-iceberg/src/main/java/io/prestosql/iceberg/HiveTableOperations.java

```diff
@@ -0,0 +1,228 @@
+    private static final StorageFormat STORAGE_FORMAT = StorageFormat.create(
+            LazySimpleSerDe.class.getName(),
+            FileInputFormat.class.getName(),
+            FileOutputFormat.class.getName());
+
+    private final HiveMetastore metastore;
+    private final String database;
+    private final String tableName;
+    private final Optional<String> owner;
+    private final Optional<String> location;
+
+    public HiveTableOperations(Configuration configuration, HiveMetastore metastore, String database, String table)
+    {
+        this(configuration, metastore, database, table, Optional.empty(), Optional.empty());
+    }
+
+    public HiveTableOperations(Configuration configuration, HiveMetastore metastore, String database, String table, String owner, String location)
+    {
+        this(configuration,
+                metastore,
+                database,
+                table,
+                Optional.of(requireNonNull(owner, "owner is null")),
+                Optional.of(requireNonNull(location, "location is null")));
+    }
+
+    private HiveTableOperations(Configuration configuration, HiveMetastore metastore, String database, String table, Optional<String> owner, Optional<String> location)
+    {
+        super(configuration);
+        this.metastore = requireNonNull(metastore, "metastore is null");
+        this.database = requireNonNull(database, "database is null");
+        this.tableName = requireNonNull(table, "table is null");
+        this.owner = requireNonNull(owner, "owner is null");
+        this.location = requireNonNull(location, "location is null");
+    }
+
+    @Override
+    public TableMetadata refresh()
+    {
+        if (location.isPresent()) {
+            refreshFromMetadataLocation(null);
+            return current();
+        }
+
+        Table table = getTable();
+
+        if (!isIcebergTable(table)) {
+            throw new PrestoException(NOT_SUPPORTED, "Not an Iceberg table: " + getSchemaTableName());
+        }
+
+        String metadataLocation = table.getParameters().get(METADATA_LOCATION_PROP);
+        if (metadataLocation == null) {
+            throw new PrestoException(HIVE_INVALID_METADATA, format("Table is missing [%s] property: %s", METADATA_LOCATION_PROP, getSchemaTableName()));
+        }
+
+        refreshFromMetadataLocation(metadataLocation);
+
+        return current();
+    }
+
+    @Override
+    public void commit(TableMetadata base, TableMetadata metadata)
+    {
+        // if the metadata is already out of date, reject it
+        if (!Objects.equals(base, current())) {
+            throw new CommitFailedException("Cannot commit: stale table metadata for %s", getSchemaTableName());
+        }
+
+        // if the metadata is not changed, return early
+        if (Objects.equals(base, metadata)) {
+            return;
+        }
+
+        String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+
+        // TODO: use metastore locking
```

Comment:
FYI the Hive table locking that is used in the Iceberg version of `HiveTableOperations` unfortunately isn't supported by Glue so possibly some alternative locking mechanism would have to be used to support Glue. I wrote about this a bit here - https://github.com/Netflix/iceberg/issues/112